File: Java/MIMIC/src/ejemplosencillo/EDA/OneMax.java
package ejemplosencillo.EDA;

import ejemplosencillo.base.IFuncionObjetivo;
import java.util.LinkedList;

public class OneMax implements IFuncionObjetivo{

    @Override
    public Integer evaluar(LinkedList<Byte> values) {
        byte suma = 0;
        for (Byte item: values){
            suma += item;
        }
        return values.size()-suma;
    }

}

File: Java/MIMIC/src/ejemplosencillo/EDA/OneMax.kt
package ejemplosencillo.EDA

import ejemplosencillo.base.IFuncionObjetivo
import java.util.LinkedList

class OneMax : IFuncionObjetivo {

    @Override
    fun evaluar(values: LinkedList<Byte>): Integer {
        var suma: Byte = 0
        for (item in values) {
            suma += item.toByte()
        }
        return values.size() - suma
    }

}
